### PR TITLE
Add default timeout for requests to the controller

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: router/xgress/options.go [hunk 1]
```diff
@@ -46,6 +46,9 @@ type Options struct {
 	MaxCloseWait time.Duration
 
 	GetSessionTimeout time.Duration
+
+	// default timeout to use when sending requests to the controller
+	ControllerTimeout time.Duration
```
Comment (michaelquigley) [inline router/xgress/options.go:51]: Where is this used?

**Review by michaelquigley — APPROVED** [review]

File: router/xgress/options.go (hunk 1 above)
Comment (plorenz) [reply router/xgress/options.go:51]: Refactoring xgress_edge to use it. xgress/request could be refactored to use it as well at some point.

Comment (michaelquigley) [reply router/xgress/options.go:51]: If it's in `openziti/fabric`, shouldn't we refactor something in `fabric` to use it?

Comment (plorenz) [reply router/xgress/options.go:51]: I may need to move it. The easier thing to refactor to use it doesn't have access xgress.Options. Maybe need to make it lop level router options, though not sure those are accessible in xgress implementations.

Comment (michaelquigley) [reply router/xgress/options.go:51]: Refactoring something like `xgress_proxy` to use `xgress.Options` doesn't seem like it would be a big deal?

**Pull request merged**